### DISCUSSION TITLE
Prevent sleep by AssistantDelay

### DISCUSSION THREAD

**Discussion by Vladislavik** [tribalfs/GalaxyMaxHzPub#150] [General]
After installing GMH on Android 14 Galaxy S24Ultra i see battery drain and on XDA forum i read comments about battery drain cosed by GMH, can you check it please, https://xdaforums.com/t/official-battery-life-thread.4552597/page-54 starting from post 1071

**Comment by tribalfs** [comment]
The app is not the cause . It has nothing to do with AssistantDelay wakelock

**Reply by Vladislavik** [reply]
When I remove GHM this wakelock by Assistant Delay removed. On forum users uninstall GHM and they see difference too

**Reply by tribalfs** [reply]
Please check my reply :
https://xdaforums.com/t/official-battery-life-thread.4552597/post-89318760

**Reply by JubilantJerry** [reply]
How do I access the source code you attached on that thread, the one that shows `NotificationManagerService` and `mWakeLockForAssistantDelay`? I found that a lot of apps cause `Prevent Sleep by AssistantDelay`, it seems like a fundamental problem with how Samsung handles apps that perform periodic updates in the background. I think Galaxy Max Hz was just a victim because I imagine it periodically sets the screen refresh rate in the background. I mentioned more about my speculation in https://xdaforums.com/t/android-os-services.4666407/

**Reply by tribalfs** [reply]
It's decompiled from framework files of a samsung oneui rom.  This one is particularly from services.jar

That wakelock is used by edge lighting and  apps/services that intercepts notifications mostly to either disable it and/or to show its own version.
